### ISSUE TITLE
Could you provide the visualization code of the affinity map in Fig. 4?

### ISSUE THREAD

**Issue by SuleBai** [wysoczanska/clip_dinoiser#9]
Hi! Thanks for your great work!

Could you provide the visualization code of the affinity map in Fig. 4?

Thanks again.

**Comment by wysoczanska** [comment]
Hi! Thanks for your interest! You can use the code below to get the affinity map from Fig. 4:
On the loaded model, you can run (you can find how to load the model in `demo.ipynb`):
`
img = Image.open(file_path).convert('RGB')
img_tens = T.PILToTensor()(img).unsqueeze(0).to(device) / 255.
with torch.no_grad():
   _, corrs_out, clip_proj_feats = model.forward_pass(img_tens)
   fig, ax = plt.subplots(1, 3)
   seed_id = 50 # you can change the feature seed here
   corrs_vis = corrs_out.clone()
   col = ax[0].imshow(corrs_vis[0][seed_id].cpu())
   ax[0].set_title('CLIP-DINOiser predicted corrs')
   fig.colorbar(col)

   dino_corrs = model.get_dino_corrs(img_tens)
   col1 = ax[1].imshow(dino_corrs[0][seed_id].cpu())
   ax[1].set_title('DINO corrs')
   fig.colorbar(col1)

   clip_proj_feats_norm = clip_proj_feats/clip_proj_feats.norm(dim=1)
   clip_proj_feats_norm = clip_proj_feats_norm.flatten(-2, -1)
   clip_corrs = torch.matmul(clip_proj_feats_norm.permute(0, 2, 1), clip_proj_feats_norm[0]).reshape(1, -1, 28, 28)
   col2 = ax[2].imshow(clip_corrs[0][seed_id].cpu())
   ax[2].set_title('MaskCLIP corrs')
   fig.colorbar(col2)
`

Let me know if it works for you!

**Comment by SuleBai** [comment]
Thanks a lot! Besides, may I ask how do you plot the actual seed(the red point in Fig. 4) on the affinity_map? 

Thanks again.

**Comment by wysoczanska** [comment]
Sure! I usually use the following:
`plt.scatter(x=seed_id % wf, y=seed_id // wf  , c='red',marker='s', s=120) `

In the above code snippet, it should work as:
`ax[0].scatter(x=seed_id % wf, y=seed_id // wf  , c='red',marker='s', s=120) `

**Comment by SuleBai** [comment]
Thanks! Appreciate to your work!
